### ISSUE TITLE
Cache not invalidating(?) when using blobfuse2 on AKS v1.25.6

### ISSUE THREAD

**Issue by fpgmaas** [Azure/azure-storage-fuse#1171]
### Which version of blobfuse was used?

```sh
blobfuse2 version 2.0.3
```

### Which OS distribution and version are you using?

AKS with Kubernetes [v1.25.6](https://github.com/kubernetes/kubernetes/releases/tag/v1.25.6)

### If relevant, please share your mount command.

I use a PersistentVolume with a PersistentVolumeClaim, which are then used in a Helm chart for Airflow.

`pv-dags.yaml`:

```yaml
apiVersion: v1
kind: PersistentVolume
metadata:
  name: pv-airflow-dags-dev
  labels:
    type: local
spec:
  capacity:
    storage: 10Gi
  accessModes:
    - ReadWriteMany
  persistentVolumeReclaimPolicy: Retain  # If set as "Delete" container would be removed after pvc deletion
  storageClassName: azureblob-fuse-premium
  mountOptions:
    - -o allow_other
    - --file-cache-timeout-in-seconds=120
  csi:
    driver: blob.csi.azure.com
    readOnly: false
    # volumeid has to be unique for every identical storage blob container in the cluster
    # character `#` is reserved for internal use and cannot be used in volumehandle
    volumeHandle: airflow-dags-dev-1
    volumeAttributes:
      resourceGroup: global-rg
      storageAccount: cadranglobal
      containerName: airflow-dags-dev
      protocol: fuse2
    nodeStageSecretRef:
      name: storage-account-credentials
      namespace: airflow-dev
```

`pvc-dags.yaml`:

```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: pvc-airflow-dags-dev
spec:
  storageClassName: azureblob-fuse-premium
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 1Gi
  volumeName: pv-airflow-dags-dev
```

### What was the issue encountered?

Files seem to be cached for way longer (indefinitely even?) than the two minutes specified in the configuration.

Let's say I have a file `airflow-dags-dev/example/memory.py ` that contains the text:

```
[...]
GB_MEMORY_USAGE_PER_POD = 2
[...]
``` 

If I make a change that to `GB_MEMORY_USAGE_PER_POD = 3`and upload that to Blob, then wait five minutes, then navigate into my pod that uses the PVC, I see:

```sh
k -n airflow-dev exec -it airflow-scheduler-<xxx>/bin/bash
cd /opt/airflow/dags/example
cat memory.py

[...]
 GB_MEMORY_USAGE_PER_POD = 2
[...]
 ```

Whereas If I inspect the Blob directly:

```sh
 az storage blob download --blob-url https://<xxx>.blob.core.windows.net/airflow-dags-dev/example/memory.py --auth-mode login

[...]
    GB_MEMORY_USAGE_PER_POD = 3
[...]
```

So the change it not reflected in the mounted directory.

If however I create a new file in blob storage it directly appears in the pod. The other way around works as well; if I create a new file in the pod it appears in the Blob storage.

### Have you found a mitigation/solution?

No, I have not found a solution for my problem yet. The issue seems similar to [this issue](https://github.com/Azure/azure-storage-fuse/issues/1000). I hoped the combination of Kubernetes v1.25.6 with the config setting `protocol: fuse2` would solve my issue, but it did not.

I have tried with a combination of settings, e.g.:

```  
mountOptions:
    - -o allow_other
    - -o attr_timeout=0
    - -o entry_timeout=0
    - -o negative_timeout=0
    - --use-attr-cache=false
    - --file-cache-timeout-in-seconds=0
    - --cache-size-mb=0
```

But I have not been successful in solving the issue. I am not sure if it's a user error or something going wrong with `blobfuse2`.

Please let me know if I can provide any more information.

**Comment by vibhansa-msft** [comment]
Are you updating the file from same path where blobfuse is mounted ? 
If no, this is not due a bug in blobfuse but rather kernel caching the file-contents. If you clear kernel page cache manually you will get the updated contents on read.
There is a work-in-progress for such use-cases to disable kernel caching on blobfuse mount (direct_io mode) and it will be available in next release.

**Comment by vibhansa-msft** [comment]
"-o direct_io" cli parameter will disable kernel caching from next release. Fix has been merged to main line and will be available in 2.0.4. Feel free to reopen if the issue is still hit in next release (2.0.4)
